astropy/io/misc/tests/test_hdf5.py

```diff
@@ -871,3 +871,26 @@ def test_round_trip_masked_table_default(tmpdir):
         t[name].mask = False
         t2[name].mask = False
         assert np.all(t2[name] == t[name])
+
+
+@pytest.mark.skipif('not HAS_YAML or not HAS_H5PY')
+def test_overwrite_serialized_meta():
+    # This used to cause an error because the meta data table
+    # was not removed from the existing file.
+
+    with h5py.File('test_data.h5', 'w', driver='core', backing_store=False) as out:
```

Comment:
Ah, I now see how you do things in memory with `backing_store=False`. That seems reasonable.

Reply:
I did not know about that until a few days ago when I saw the way astropy uses StringIO for some tests. Since keeping everything in memory seemed a good idea, I went out searching for something similar to HDF5 files.